internal/users/cache/db_test.go

```diff
@@ -60,6 +60,14 @@ func TestNew(t *testing.T) {
 			if tc.perm != nil {
 				err := os.Chmod(dbDestPath, *tc.perm)
 				require.NoError(t, err, "Setup: could not change mode of database file")
+
+				if *tc.perm == perm0644 {
+					currentUser, err := user.Current()
+					require.NoError(t, err)
+					if currentUser.Username == "root" {
+						t.Skip("Can't do permission checks as root")
+					}
+				}
```

Comment:
We already talked about this before, but I'll comment here as well: We never did something like this in our other projects and these tests are important and were passing in Launchpad, so skipping them is not fixing anything.

Reply:
If those tests are not reliable when running as root, we remove them. As Denis said, we don’t want to silently skip some tests under some conditions. This is how we then end up with tests never asserting anything and we think everything is going alright because of the global "PASS". So it’s safer from a distro POV, but it doesn’t help the distro either.

Reply:
Well I don't think it would be correct to remove them either... We're doing a test that wants to ensure that we may return errors in specific conditions, but this is not true all the times.

Now, not having the tests implies that we don't want to cover at least some cases in which we're getting such errors... But I think it's sane doing it.

We've dozen of tests like this in many places when doing lowlevel operations that may not work in some specific conditions, and when this is the case, I think is sane to ignore them, but not all the times.

However, I'm fine with any decision. But skipping a test doesn't seem the end of the world (given that we do it already in other cases in which the tests prerequisites aren't met).

Reply:
In our projects, the rule is to skip tests on environment variable set. That way, it’s explicit, but not based on $condition_on_running in the machine. Explicit is always better than implicit.

For instance, in adsys, we skip integration tests using a docker image when building the package, but this is controlled with an env variable.

Reply:
I see, having an env variable here is also possible, something like `AUTHD_TESTS_SKIP_PERMISSION_CHECKS` when building the deb?

Reply:
Ack done